src/goto-programs/goto_convert.cpp

```diff
@@ -474,12 +474,16 @@ bool goto_convertt::rewrite_vla_decl_size(exprt &size, goto_programt &dest)
     return true;
   }
 
+  // Constant size is not a VLA
+  if(size.is_constant())
+  {
+    return false;
+  }
+
   // We have to replace the symbol by a temporary, because it might
   // change its value in the future
   // Don't create a symbol for temporary symbols
-  if(
-    size.is_symbol() &&
-    size.identifier().as_string().find("tmp$") == std::string::npos)
+  if(size.identifier().as_string().find("tmp$") == std::string::npos)
```

Comment:
Why did you remove the `is_symbol()` check?

Reply:
The issue was that `int foo[a+1]` had a size of type `add` instead of a `symbol`, so it was treated as a constant array.

Reply:
Right, two things:
- In the old irep if we call an accessor that was not allocated, it will allocate a new nil irep and return it, e.g., calling `identifier()` in an `add` irep will allocate memory for that accessor in the add object.
- At this point we know that this is not a constant anymore because of the new guard on line 478, I think the code here should be something like:
```
  // Constant size is not a VLA
  if(size.is_constant())
    return false;

  // We have to replace the symbol by a temporary, because it might
  // change its value in the future
  // Don't create a symbol for temporary symbols
  if(
    size.is_symbol() &&
    size.identifier().as_string().find("tmp$") == std::string::npos)
  {
    // Old size symbol
    exprt old_size = size;

    // Replace the size by a new variable, to avoid wrong results
    // when the symbol used to create the VLA is changed
    size = symbol_expr(new_tmp_symbol(size.type()));

    codet assignment("assign");
    assignment.reserve_operands(2);
    assignment.copy_to_operands(size);
    assignment.copy_to_operands(old_size);
    assignment.location() = old_size.location();
    copy(assignment, ASSIGN, dest);
  }

  return true;
}


```